sources/packages/api/src/services/application/application.service.ts

```diff
@@ -675,35 +685,133 @@ export class ApplicationService extends RecordDataModelService<Application> {
 
   /**
    * Get all active applications of an institution location
-   * with application_status is completed
+   * with application status completed and respective archive status.
    * @param locationId location id .
+   * @param paginationOptions options to execute the pagination.
+   * @param applicationStatus View status of applications requested by user.
    * @returns Student Active Application list.
    */
-  async getActiveApplications(locationId: number): Promise<Application[]> {
+  async getActiveApplications(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+    archived: boolean,
+  ): Promise<PaginatedResults<Application>> {
     // TODO: there are two similar methods to get one and many records for the same list/details getActiveApplication and getActiveApplications. Can we use only one?
-    return this.repo
+    const activeApplicationQuery = this.repo
       .createQueryBuilder("application")
       .select([
         "application.applicationNumber",
         "application.id",
         "currentAssessment.id",
-        "application.applicationStatus",
         "offering.studyStartDate",
         "offering.studyEndDate",
         "student.id",
         "user.firstName",
         "user.lastName",
+        "application.isArchived",
+        "studentScholasticStanding.id",
       ])
       .leftJoin("application.currentAssessment", "currentAssessment")
       .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "currentAssessment.studentScholasticStanding",
+        "studentScholasticStanding",
+      )
       .innerJoin("application.student", "student")
       .innerJoin("student.user", "user")
       .where("application.location.id = :locationId", { locationId })
       .andWhere("application.applicationStatus = :applicationStatus", {
         applicationStatus: ApplicationStatus.completed,
-      })
-      .orderBy("application.applicationNumber", "DESC")
-      .getMany();
+      });
+
+    activeApplicationQuery.andWhere("application.isArchived = :isArchived", {
```

Comment:
this could be a continuation to line 725 as it is not used conditionally.

Reply:
Moving it away from here will return wrong results. Basically, this filter needs to execute when the application is either archived or not (true or false). If application is archived (Unavailable), then another filter needs to be executed on applications with where scholastic standing id is not null (Completed). Unavailable and Completed applications need to be displayed within the same query.

Reply:
@henryoforeh-aot the first comment from @dheepak-aot still makes sense.
Unless `isArchived` is optional does not make sense to keep it separated from the main query.
IMO it could be optional, but right now it is not.

Reply:
I can see that this method is still not changed. Any particular reason?

Reply:
I can see it is changed now. It looks good to me.